ref: add config --project flag and standardize the others (#2064)
iterative/dvc.org: content/docs/command-reference/cache/dir.md

## Synopsis

```usage

> usage: dvc cache dir [-h] [--global | --system | --repo | --local] [-u] [value]

Author: Maybe `--repo` -> `--project` (similar to `gc`'s `--projects`, but that has a different meaning though), but I like the repo more, tbh.

Reviewer: Let me understand what the option means exactly so can opine... ⌛

Reviewer: I vote for `--project`.

- Repo sounds like Git.
- We already use the term "project" more for this concept (in docs).

Reviewer: BUT, it's not obvious (from the name) what the difference is between "repo/project" and "local". Isn't the repo/project local? It wasn't an issue before since there was no `--repo` flag but I wonder if we should also rename `--local` to `--private` or something.

Author: In things like `dvc import/get` we use `DVC or Git repository` and not `project`. The term `project` seems a bit synthetic.

Regarding `--private` we have https://github.com/iterative/dvc/issues/3670 , but not sure it will be done for 2.0

Reviewer: > Regarding --private we have iterative/dvc#3670

Thanks for reminding me about that one, OK cool no need to worry about it here then.

Reviewer: I don't think "project" is synthetic. Maybe just a bit generic, like anything you're working on can be a project. But that's why when we need to be very explicit we can say "DVC project".

Repo basically implies Git and DVC does feature --no-scm mode so there are non-repo DVC project. That' the problem I see with "repo"... Plus let's think about where that comes from: *data repository* OK, does it work for *data repository*? I think that term is already sort of established in the subfield of data mining.

That said we currently have this terminology problem in docs so it's not like I have a definitive answer. But I guess whatever we decide here I can apply throughout (later).

Cc @shcheklein

Reviewer: \* *code repository* OK, does it work for *data repository*?

Reviewer: p.s.

> In things like dvc import/get we use DVC or Git repository

That is correct because import/get don't work with --no-scm projects, only repos.

Author: Let it be --project, I'm ok with it. I'll adjust the code and docs soon.

Author: @dberenbaum What do you think about `--repo` vs `--project`?

Reviewer 2: `--project` works for me. I don't have a strong opinion as long as we try to maintain consistency.

Author: https://github.com/iterative/dvc/pull/5399

> usage: dvc cache dir [-h] [--global | --system | --project | --local] [-u] [value]

Reviewer: Please wrap at 72 chars e.g.:
```suggestion
usage: dvc cache dir [-h] [--global | --system | --project | --local]
                     [-u] [value]
```

Reviewer: ☝️ generalizable too

**relative to the config file location**. However, if the `value` provided is an
absolute path, then it's preserved as it is.

> When reading (`value` is not provided) the value is read from the system,
> global, repo and local configuration files by default, and options `--system`,
> `--global`, `--repo`, and `--local` can be used to tell the command to read from
> only that location.

Reviewer: ```suggestion
If no path `value` is provided to this command, it prints the path for current
cache directory. the value is searched in the system, global, repo, and local
config files by default (in that order). Options `--system`, `--global`,
`--repo`, and `--local` can be used to look only in one place.
```

Reviewer: > Options `--system`, `--global`, `--repo`, and `--local` can be used to look only in one place.

If we want to simplify the description, we can skip this since it will be explained in the **Options** section.

Reviewer: > since it will be explained

Actually, I see it's not exactly specified there, but it could (probably should) — again, to avoid overcomplicating the description.

Author: The original text is adapted from git docs, but ok, let's have it this way.

Reviewer: Thanks. I don't think git docs are usually super relevant but... that can definitely be a great starting point 👍

> When writing (`value` is provided or `--unset` is used), the new value is
> written to the projects configuration file by default (`.dvc/config`), and
> options `--system`, `--global` and `--local` can be used to tell the command to
> write to that location (you can say `--repo` but that is the default).

Reviewer: I would move this up as part of the first paragraph or as a 2nd paragraph. Suggested text:
```suggestion
The `value` is written to the repo's config file by default (`.dvc/config`).
Options `--system`, `--global`, and `--local` can be used to write to the other
possible config files.
```

Reviewer: > Options `--system`, `--global` ...

Same

Author: Again, adapted from git help, but ok let's tweak it.

## Options

- `--global` - use a global config file (e.g. `~/.config/dvc/config`) instead of

>   the project's `.dvc/config`.

Reviewer: For example here we're still calling it "project". We should probably be consistent to avoid confusions. Repo and project don't always refer to the same thing (depends on the context).

- `--system` - use a system config file (e.g. `/etc/dvc/config`) instead of
  `.dvc/config`.

> - `--repo` - use a project's config file (`.dvc/config`).

Reviewer: Here's an example of explaining about changing read/write behavior with these:

```suggestion
- `--repo` - use the regular config file (`.dvc/config`) only when reading the
  value. Has no effect when writing (this is the default).
```

absolute path, then it's preserved as it is.

> If no path `value` is provided to this command, it prints the path for current

Reviewer: ```suggestion
If no path `value` is given, it prints the current path of the
```
> Will need rewrapping with the remainder of the paragraph

> cache directory. The value is searched in the system, global, project, and local
> config files by default (in that order). Options `--system`, `--global`,
> `--project`, and `--local` can be used to look only in one place.

Reviewer: ```suggestion
cache directory. The value is read from the system, global, project, and local
config files (in that order). The `--system`, `--global`, `--project`, and
`--local` options can be used to read from that configuration only.
```
> Will need to rewrap along with the previous line...

> The `value` is written to the project's config file by default (`.dvc/config`).
> Options `--system`, `--global`, and `--local` can be used to write to the other
> possible config files.

Reviewer: ```suggestion
If a `value` is given, it's written to the project-level config file by default
(`.dvc/config`). Options `--system`, `--global` and `--local` can be used to
write to that location instead.
```

## Options

> - `--global` - use a global config file (e.g. `~/.config/dvc/config`) instead of

Reviewer: Please try to reuse the option texts from my previous review here and in all other places if possible.
